src/core/lib/security/security_connector/local/local_security_connector.cc

```diff
@@ -124,19 +129,51 @@ static grpc_security_status local_auth_context_create(grpc_auth_context** ctx) {
 }
 
 static void local_check_peer(grpc_security_connector* sc, tsi_peer peer,
+                             grpc_endpoint* ep,
                              grpc_auth_context** auth_context,
                              grpc_closure* on_peer_checked) {
+  int fd = grpc_endpoint_get_fd(ep);
+  grpc_resolved_address resolved_addr;
+  resolved_addr.len = static_cast<socklen_t>(sizeof(struct sockaddr_storage));
+  bool is_endpoint_ok = false;
+  if (getsockname(fd, reinterpret_cast<grpc_sockaddr*>(resolved_addr.addr),
+                  &resolved_addr.len) == 0) {
+    grpc_sockaddr_in* addr4 =
+        reinterpret_cast<grpc_sockaddr_in*>(resolved_addr.addr);
+    grpc_sockaddr_in6* addr6 =
+        reinterpret_cast<grpc_sockaddr_in6*>(resolved_addr.addr);
+    // UDS
```

Comment:
grpc_local_connect_type in credential is not really used to check endpoint. We want to check endpoint is UDS and local_connect_type in base.channel_creds is also UDS to make sure they are consistent. Same for local TCP connections.

Reply:
Good point. The change is applied.